linkerd/announcer/curatorsd/src/main/scala/io/buoyant/linkerd/announcer/curatorsd/CuratorSDAnnouncer.scala

```diff
@@ -0,0 +1,95 @@
+package io.buoyant.linkerd.announcer.curatorsd
+
+import java.net.InetSocketAddress
+
+import com.twitter.finagle.{Announcement, Path}
+import com.twitter.logging.Logger
+import com.twitter.util.{Future, Time}
+import io.buoyant.linkerd.FutureAnnouncer
+import org.apache.curator.framework.CuratorFrameworkFactory
+import org.apache.curator.retry.ExponentialBackoffRetry
+import org.apache.curator.x.discovery._
+import org.apache.curator.x.discovery.details.{JsonInstanceSerializer, ServiceDiscoveryImpl}
+import org.apache.log4j.{BasicConfigurator, Level}
+
+/**
+ * Announcer that uses the curator service discovery format.
+ * <p>
+ * Format used for the service name Path: /#/io.l5d.curatorsd/{full service name}[#{tenant}]
+ * tenant is optional. If it's empty or the last part is missing, it will be registered as a multi-tenant service.
+ */
+class CuratorSDAnnouncer(zkConnectStr: String) extends FutureAnnouncer {
+
+  override val scheme: String = "zk-curator"
+
+  private val log = Logger("curator")
+
+  private val DefaultBaseSleepTimeMs = 1000;
+  private val DefaultMaxRetries = 3
+
+  private val curatorClient = CuratorFrameworkFactory.builder
```

Comment:
I am not sure how curator handles disconnection from zookeeper. If it is disconnected from zookeeper can you still use stale routing data in your client.